controllers/ingressnodefirewall_controller.go

```diff
@@ -104,16 +104,18 @@ func (r *IngressNodeFirewallReconciler) Reconcile(ctx context.Context, req ctrl.
 		}
 
 		// If the node name was found ...
-		// a) compare the specs. If the specs are different, update the current spec.
-		if !equality.Semantic.DeepEqual(ingressNodeFirewallCurrentNodeState.Spec, desiredNodeState.Spec) {
+		// a) compare the specs or owner reference. If the specs or owner reference are different, update the current spec.
+		if !equality.Semantic.DeepEqual(ingressNodeFirewallCurrentNodeState.Spec, desiredNodeState.Spec) ||
+			!equality.Semantic.DeepEqual(ingressNodeFirewallCurrentNodeState.OwnerReferences, desiredNodeState.OwnerReferences) {
```

Comment:
can u explain in more details about the race condition and what does it have to do with OwnerReferences ?